lib/honeybadger/plugins/active_job.rb

```diff
@@ -0,0 +1,29 @@
+module Honeybadger
+  module Plugins
+    module ActiveJob
+            
+      Plugin.register {
+        requirement { defined?(::Rails.application) && ::Rails.application }
+        requirement { 
+          ::Rails.application.config.active_job[:queue_adapter] == :async
+        }
+        
+        execution {
+          ::ActiveJob::Base.class_eval do |base| 
+            base.set_callback :perform, :around do |param, block|
+              begin
+                Honeybadger.flush {
```

Comment:
Using flush here means that any exceptions reported via `Honeybadger.notify` from inside the block call will be reported before continuing. The exception reported in the outer `rescue` statement on line 19 below will also be reported immediately because of the `sync: true` option.

Reply:
Is this the intended behavior though (to report immediately)?

Reply:
Good question. Not sure **I** can answer that ;). I think from what I've read most of the time the intention is to make testing easier but I guess wrapping the test code in `.flush {}` should be sufficient.

Reply:
In general, `Honeybadger.flush` is necessary in cases where the process or thread could be killed before the agent finishes reporting error in the worker queue.